clientv3/client.go

```diff
@@ -282,30 +296,39 @@ func (c *Client) ActiveConnection() *grpc.ClientConn {
 }
 
 // retryConnection establishes a new connection
-func (c *Client) retryConnection(err error) (newConn *grpc.ClientConn, dialErr error) {
+func (c *Client) retryConnection(err error) {
+	oldconn := c.conn
+	defer func() {
```

Comment:
can we put the defer after c.mu.lock? it seems strange to have c.mu.unlock before c.mu.lock